### PR TITLE
Qualitative/BioGrid-based predictions of regulatory relationships

### PR DESCRIPTION
**Pull request by NElnour**
These tools are for BioGrid-based predictions of regulatory relationships using an interaction interpretation map developed on both GGI and PPI.

Note: someone used my fork to build their tool, so there are R functions here towards that end.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -24,6 +28,10 @@ LazyData: true
 biocViews: Software, Infrastructure
 Imports:
     Biostrings,
+    utils,
+    stats,
```
Comment (sapirl) [inline DESCRIPTION]: No need to add base R packages

File: R/getGeneticInteractome.R [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# getGeneticInteractome.R
+#' @title getGeneticInteractome
+#' @author Nada Elnour, \email{nada.elnour@@mail.utoronto.ca}
+#'
+#' \code{getGeneticInteractome} augments systems with BioGRID genetic
+#' interaction annotations
+#'
+#' @param mySys The dataframe of PPI between system's components
```
Comment (sapirl) [inline R/getGeneticInteractome.R]: The @param and @return should match the format described here
https://github.com/hyginn/BCB420.2019.ESA/blob/master/PullRequestReviewNotes.md

File: R/getGeneticInteractome.R [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+#' @param criterion A string, either "stringent" or "relaxed", specifying
+#' whether only GGI between physical interactors should be selected.
+#'
+#' @return The dataframe of system components and either their GGI between
+#' physical interactors should be selected (iff \code{criterion} == "stringent")
+#' or all GGI of physical interactors (iff \code{relaxed} == "stringent")
+#'
```
Comment (sapirl) [inline R/getGeneticInteractome.R:18]: @example + short explanation is needed for every exported function

File: R/hypothesize.R [hunk 4]
```diff
@@ -0,0 +1,93 @@
+# hypothesize.R
+#' @title hypothesize
+#' @author Nada Elnour, \email{nada.elnour@@mail.utoronto.ca}
+#'
+#' \code{hypothesize} generates an annotated hypothesis graph whose nodes are
+#' components of \code{mySys} and edges are GGI interpretations in either
+#' \code{EMAP} or \code{GMAP}
+#'
+#' @param network A dataframe of physically-interacting system components with
```
Comment (sapirl) [inline R/hypothesize.R]: As above: changing the parameters and results according to this format
https://github.com/hyginn/BCB420.2019.ESA/blob/master/PullRequestReviewNotes.md

File: R/hypothesize.R [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+#' @title visualizeInteractions
+#' @author Nada Elnour, \email{nada.elnour@@mail.utoronto.ca}
+#'
+#' A helper function of \code{hypothesize} that plots the hypothesis graph
+#'
+#' @inheritParams hypothesize
+#' @param gmap The dataframe obtained from \code{makeGMAP}.
+#' @param emap The dataframe obtained from \code{makeEMAP}.
+#'
+visualizeInteractions <- function(network, emap, ppi_ggi, gmap) {
```
Comment (sapirl) [inline R/hypothesize.R:52]: The return value needs to be explicit, by using return()

File: R/hypothesize.R [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  }
+
+#' @title visualizeInteractions
+#' @author Nada Elnour, \email{nada.elnour@@mail.utoronto.ca}
+#'
+#' A helper function of \code{hypothesize} that plots the hypothesis graph
+#'
+#' @inheritParams hypothesize
+#' @param gmap The dataframe obtained from \code{makeGMAP}.
+#' @param emap The dataframe obtained from \code{makeEMAP}.
```
Comment (sapirl) [inline R/hypothesize.R]: The @return is missing.

Also, as above, changing the parameters format

File: R/makeMaps.R [hunk 7]
```diff
@@ -0,0 +1,110 @@
+# makeMaps.R
+
+#' Generate the genetic interpretation map of BioGrid GGI tags of system's physical interactions.
+#'
+#' @return The dataframe mapping BioGrid GGI tag to its interpretation assuming that the system's components also interact physically.
```
Comment (sapirl) [inline R/makeMaps.R]: Needs to match the format
https://github.com/hyginn/BCB420.2019.ESA/blob/master/PullRequestReviewNotes.md

File: R/makeMaps.R [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+                     effect = effects,
+                     notes = notes, stringsAsFactors = FALSE)
+
+  return(EMAP)
+
+}
+
+#' Generate the genetic interpretation map of BioGrid GGI tags.
+#'
+#' @return The dataframe mapping BioGrid GGI tag to its interpretation.
```
Comment (sapirl) [inline R/makeMaps.R]: As above

File: R/predictDirectedNetworks.R [hunk 9]
```diff
@@ -0,0 +1,63 @@
+# predictDirectedNetworks.R
+
+source('./R/fetchData.R', echo=TRUE)
```
Comment (sapirl) [inline R/predictDirectedNetworks.R]: No need to source those files, as they are installed and loaded whenever as part of the package

File: R/predictDirectedNetworks.R [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+source('./R/SyDButils.R', echo=TRUE)
+# ==============================================================================
+#' @title getSysInteractions
+#' @author Nada Elnour, \email{nada.elnour@@mail.utoronto.ca}
+#'
+#' \code{getSysInteractions} filter system by physically-interacting components
+#' and return their genetic interactions
+#'
+#' @param sysName A string specifying the full path to the excel sheet
```
Comment (sapirl) [inline R/predictDirectedNetworks.R]: As above: format

File: R/predictDirectedNetworks.R [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+#'
+#' @return A dataframe of system components and either their GGI between
+#' physical interactors should be selected (iff \code{criterion} == "stringent")
+#' or all GGI of physical interactors (iff \code{relaxed} == "stringent")
+#'
+#' @importFrom dplyr filter
+#' @import utils
+#' @include SyDButils.R fetchData.R
+#' @importFrom stats complete.cases
+#' @examples
```
Comment (sapirl) [inline R/predictDirectedNetworks.R:23]: A short explanation is required for each example of an exported function

**Review by sapirl — CHANGES_REQUESTED** [review]
The comments

**Review by sapirl — CHANGES_REQUESTED** [review]
The changes

File: R/predictDirectedNetworks.R (hunk 10 above)
Comment (NElnour) [reply R/predictDirectedNetworks.R]: Fixed.

File: R/predictDirectedNetworks.R (hunk 11 above)
Comment (NElnour) [reply R/predictDirectedNetworks.R:23]: Fixed.

File: R/predictDirectedNetworks.R (hunk 9 above)
Comment (NElnour) [reply R/predictDirectedNetworks.R]: Fixed.

File: R/makeMaps.R (hunk 7 above)
Comment (NElnour) [reply R/makeMaps.R]: Fixed.

File: R/hypothesize.R (hunk 6 above)
Comment (NElnour) [reply R/hypothesize.R]: fixed.

File: R/hypothesize.R (hunk 5 above)
Comment (NElnour) [reply R/hypothesize.R:52]: fixed...although the function itself does not return anythin; it just plots and formats the graphs'  options

File: R/hypothesize.R (hunk 4 above)
Comment (NElnour) [reply R/hypothesize.R]: fixed

File: R/getGeneticInteractome.R (hunk 3 above)
Comment (NElnour) [reply R/getGeneticInteractome.R:18]: fixed

File: R/getGeneticInteractome.R (hunk 2 above)
Comment (NElnour) [reply R/getGeneticInteractome.R]: fixed

File: DESCRIPTION (hunk 1 above)
Comment (NElnour) [reply DESCRIPTION]: fixed

File: R/hypothesize.R (hunk 6 above)
Comment (sapirl) [reply R/hypothesize.R]: Great! after fixing every thing, please submit a new pull request (that way the tag for "changes requested" will be removed)

**Review by sapirl — APPROVED** [review]

**Pull request closed without merging**